File: leetcode/100-999/125. Valid Palindrome/solution_2.java
class Solution
{
    public boolean isPalindrome(String s)
    {
        int left = 0;
        int right = s.length() - 1;

        while (left < right)
        {
            var leftCh = s.charAt(left);
            var rightCh = s.charAt(right);

            if (!Character.isLetterOrDigit(leftCh))
                left++;

            if (!Character.isLetterOrDigit(rightCh))
                right--;

            if (Character.isLetterOrDigit(leftCh) && Character.isLetterOrDigit(rightCh))
            {
                if (Character.toLowerCase(leftCh) != Character.toLowerCase(rightCh))
                    return false;
                
                left++;
                right--;
            }
        }
        
        return true;
    }
}

File: leetcode/100-999/125. Valid Palindrome/solution_2.js
/**
 * 
 * @param {string} ch 
 * @returns {string} 
 */

function isLetterOrDigit(ch) 
{
    return /^[a-z0-9]$/i.test(ch);
}

/**
 * @param {string} s
 * @return {boolean}
 */

var isPalindrome = function(s) 
{
    let left = 0
    let right = s.length - 1
    
    while (left < right)
    {
        const leftCh = s[left]
        const rightCh = s[right]

        if (!isLetterOrDigit(leftCh))
            left++;

        if (!isLetterOrDigit(rightCh))
            right--;
        
        if (isLetterOrDigit(leftCh) && isLetterOrDigit(rightCh))
        {
            if (leftCh.toLowerCase() != rightCh.toLowerCase())
                return false;

            left++;
            right--;
        }
    }

    return true;
};